packages/common/src/logger-provider.ts

```diff
@@ -1,279 +1,38 @@
-    },
-}
+// Invalid to access before calling `initializeLogger()`
+export let logger: DiagnosticsLogger
 
-/**
- * Logger options
- */
-interface Options {
-    level?: string;
-    format?: string; // [text | json]
-    stacktrace?: {
-        levels: ['trace', 'warn', 'error'];
-        depth: 3;
-        excess: 0;
-    };
-    component?: string;
+
+export interface LoggerConfig {
+    logPath?: string,
+    logLevel?: LogLevel | string,
+    logToFiles?: boolean,
 }
 
-/**
- * Plugin options
- */
-interface PluginOptions {
-    [index: string]: any;
+
+const logLevelMapping = {
+    'debug': LogLevel.debug,
+    'important': LogLevel.important,
+    'warn': LogLevel.warn
 }
 
-/**
- * Function type for plugins
- * @dev Must call `setLevel` on `rootLogger` to be enabled
- */
-type Plugin = (
-    rootLogger: log.RootLogger,
-    pluginState?: any,
-    loggerOptions?: Options,
-    pluginOptions?: PluginOptions
-) => void;
+const DEFAULT_LOG_CONFIG = {
+    logLevel: LogLevel.important,
+    logPath: path.join(os.homedir(), ".ceramic", "logs/"),
+    logToFiles: false
+}
 
 /**
  * Global Logger factory
  */
-class LoggerProvider {
-    private static options: Options;
-
-    /**
-     * Initialize root logger
-     * @param opts - Options
-     * @returns Modified options
-     */
-    static init(opts = defaultOpts): Options {
-        this.options = Object.assign(defaultOpts, opts)
-
-        if (this.options.level) {
-            log.setLevel(this.options.level as LogLevelDesc)
-        } else {
-            log.enableAll() // enable all levels (TRACE)
-        }
-
-        LoggerProvider._applyPrefix()
-        LoggerProvider._includeJsonPlugin()
-        return cloneDeep(this.options)
-    }
-
-    /**
-     * Adds `plugin` to the logger instance
-     * @param plugin Plugin function to add
-     * @param pluginState Shared state specific to `plugin`
-     * @param loggerOptions Options returned from LoggerProvider.init
-     * @param pluginOptions Options specific to `plugin`
-     */
-    static addPlugin(
-        plugin: Plugin,
-        pluginState?: any,
-        loggerOptions?: Options,
-        pluginOptions?: PluginOptions
-    ): void {
-        plugin(log, pluginState, loggerOptions, pluginOptions)
-    }
+export class LoggerProvider {
+    static init(config: LoggerConfig) {
+        config = { ...DEFAULT_LOG_CONFIG, ...config }
 
-    /**
-     * Applies prefix
-     * @private
-     */
-    static _applyPrefix(): void {
-        prefix.reg(log)
-        prefix.apply(log, {
-            format(level, name, timestamp) {
-                return LoggerProvider._toText(timestamp, level, name)
-            },
-            timestampFormatter(date) {
-                return date.toISOString()
-            }
-        })
+        const logLevel: LogLevel = typeof config.logLevel == "string" ? logLevelMapping[config.logLevel] : config.logLevel
+        logger = new DiagnosticsLogger(config.logPath, logLevel, config.logToFiles);
```

Comment:
Why does logger need to be assigned here?

Why not something like
```
export const logger = new LoggerProvider()
```

Reply:
Oh I see because the config options get passed in when we create the Ceramic instance. So would this be a problem if I import a package from the monorepo without actually starting a Ceramic instance? Then the logger would break...?

Reply:
Ok so probably better to use different loggers so that we don't require that. For example if I want to import @ceramicnetwork/ipfs I shouldn't be required to call `LoggerProvider.init()` to use it, or @ceramicnetwork/docid, etc.

Reply:
Hmm... what exactly are you proposing?  There has to be **somewhere** that handles the configuration of what file path to put the logs.  Either we have one central place, but then all packages that use the logger are dependent on that central place to configure the logger appropriately, or every package that uses the logger needs to parse the config file and configure the logging path themselves, making each package dependent on our config file format.

Reply:
So it seems like each package needs to configure its own logger. And yes need to be able to read config. I would propose env vars, unless it's the cli where we get config from commander cli flags.

Reply:
Okay, I think the core problem was trying to export a global logger instance that could be imported directly.  I changed `LoggerProvider` to be a more traditional factory so everything that needs a logger instance will need to explicitly construct what it needs with whatever config it wants.  For now only the diagnostic logger is created, but I imagine `Ceramic.create()` eventually creating specific ServiceLoggers and giving them to the services that need them.  For instance I imagine there will be an `httpLogger` that is given to the `Dispatcher` in its constructor.  My hope is that only the `DiagnosticLogger` is global-enough to be on the `Context` and the other loggers will all be service-specific.